pkg/deploy/controller/deploymentconfig/controller_test.go

```diff
@@ -1,835 +1,570 @@
-					fn: func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
-						return true, nil, kerrors.NewNotFound("imagestreamtag", "test-image-stream:"+imageapi.DefaultImageTag)
-					},
-				},
-				{
-					verb:     "list",
-					resource: "buildconfigs",
-					fn: func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
-						return true, mkBuildConfigList(), nil
-					},
-				},
-				{
-					verb:     "get",
-					resource: "imagestreams",
-					fn: func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
-						return true, nil, nil
-					},
-				},
-			},
-			expectedDetails: "The image trigger for image stream tag \"test-image-stream:latest\" will have no effect because image stream tag \"test-image-stream:latest\" does not exist.\n\tIf image stream tag \"test-image-stream:latest\" is expected, check build config \"mybc\" which produces image stream tag \"test-image-stream:latest\".",
-			expectedErr:     false,
+			name:             "(compat) scale up latest/active completed deployment",
+			replicas:         1,
+			newVersion:       2,
+			expectedReplicas: 5,
+			before: []deployment{
+				{version: 1, replicas: 0, status: deployapi.DeploymentStatusComplete, cancelled: false},
+				{version: 2, replicas: 5, status: deployapi.DeploymentStatusComplete, cancelled: false},
+			},
+			after: []deployment{
+				{version: 1, replicas: 0, replicasA: newint(0), status: deployapi.DeploymentStatusComplete, cancelled: false},
+				{version: 2, replicas: 5, replicasA: newint(5), status: deployapi.DeploymentStatusComplete, cancelled: false},
+			},
+			errExpected: false,
 		},
 		{
-			name:           "no deployments - not found image stream",
-			controller:     controller,
-			version:        1,
-			hasDeployments: true,
-			status:         deployapi.DeploymentStatusFailed,
-			reactions: []reaction{
-				{
-					verb:     "get",
-					resource: "imagestreamtags",
-					fn: func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
-						return true, nil, kerrors.NewNotFound("imagestreamtag", "test-image-stream:latest")
-					},
-				},
-				{
-					verb:     "get",
-					resource: "imagestreams",
-					fn: func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
-						return true, nil, kerrors.NewNotFound("imagestream", "test-image-stream")
-					},
-				},
-				{
-					verb:     "get",
-					resource: "imagestreams",
-					fn: func(action ktestclient.Action) (handled bool, ret runtime.Object, err error) {
-						return true, nil, kerrors.NewNotFound("imagestream", "test-image-stream")
-					},
-				},
-			},
-			expectedDetails: "The image trigger for image stream tag \"test-image-stream:latest\" will have no effect because image stream \"test-image-stream\" does not exist.",
-			expectedLatest:  true,
-			expectedErr:     false,
+			name:             "(compat) scale up active (not latest) completed deployment",
```

Comment:
How is it possible for an old client to scale up an active deployment? They would have to explicitly target the deployment, right?